Common/Product/SharedProject/CommonPackage.cs

```diff
@@ -83,20 +83,24 @@ public abstract class CommonPackage : AsyncPackage, IOleComponent {
         internal static object CommandsLock => _commandsLock;
 
         protected override void Dispose(bool disposing) {
-            _uiThread.MustBeCalledFromUIThreadOrThrow();
-            try {
-                if (_componentID != 0) {
-                    _compMgr.FRevokeComponent(_componentID);
-                    _componentID = 0;
-                }
+            if (_uiThread != null)
```

Comment:
This change feels wrong - now `Dispose()` is callable from any thread, but it won't actually dispose all the things that need to be disposed. If it's getting called from a background thread somewhere, shouldn't the caller be fixed?

Reply:
Yeah it was caused by me missing some stuff. UI thread is actuall null in some cases. I'll remove the invokeasync